ambry-cloud/src/main/java/com/github/ambry/cloud/azure/ShardedStorageClient.java

```diff
@@ -0,0 +1,258 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud.azure;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.rest.Response;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.Context;
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.BlobServiceClient;
+import com.azure.storage.blob.batch.BlobBatchAsyncClient;
+import com.azure.storage.blob.models.AccessTier;
+import com.azure.storage.blob.models.BlobHttpHeaders;
+import com.azure.storage.blob.models.BlobProperties;
+import com.azure.storage.blob.models.BlobRange;
+import com.azure.storage.blob.models.BlobRequestConditions;
+import com.azure.storage.blob.models.BlobStorageException;
+import com.azure.storage.blob.models.BlockBlobItem;
+import com.azure.storage.blob.models.DownloadRetryOptions;
+import com.azure.storage.common.policy.RequestRetryOptions;
+import com.github.ambry.cloud.CloudBlobMetadata;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.config.CloudConfig;
+import com.github.ambry.utils.Utils;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.MalformedURLException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+/**
+ * Implements AzureStorageClient interface to handle multiple Azure storage accounts.
+ */
+public class ShardedStorageClient implements AzureStorageClient {
+  public final List<StorageClient> storageClientList;
+  public final List<Integer> storageClientPartitionBoundaries;
+
+  /**
+   * Constructor for {@link ShardedStorageClient}.
+   * @param cloudConfig {@link CloudConfig} object.
+   * @param azureCloudConfig {@link AzureCloudConfig} object.
+   * @param azureMetrics {@link AzureMetrics} object.
+   * @param blobLayoutStrategy {@link AzureBlobLayoutStrategy} object.
+   */
+  public ShardedStorageClient(CloudConfig cloudConfig, AzureCloudConfig azureCloudConfig, AzureMetrics azureMetrics,
+      AzureBlobLayoutStrategy blobLayoutStrategy) throws ReflectiveOperationException {
+    List<AzureCloudConfig.StorageAccountInfo> storageAccountInfoList = azureCloudConfig.azureStorageAccountInfo;
+
+    storageClientList = new ArrayList<>();
+    storageClientPartitionBoundaries = new ArrayList<>();
+    for (AzureCloudConfig.StorageAccountInfo storageAccountInfo : storageAccountInfoList) {
+      storageClientList.add(Utils.getObj(azureCloudConfig.azureStorageClientClass, cloudConfig, azureCloudConfig,
+          azureMetrics, blobLayoutStrategy, storageAccountInfo));
+      storageClientPartitionBoundaries.add(storageAccountInfo.getPartitionRangeStart());
+    }
+    storageClientPartitionBoundaries.add(Integer.MAX_VALUE);
```

Comment:
Maybe add a log here saying something like "5 storage clients were created with the following boundaries..... ". This will help in debugging any issues.

Reply:
Done